internal/url/url.go

```diff
@@ -26,15 +39,17 @@ func ParseQuery(query string, separator byte) map[string][]string {
 		if i := strings.IndexByte(key, '='); i >= 0 {
 			key, value = key[:i], key[i+1:]
 		}
-		key = QueryUnescape(key)
-		value = QueryUnescape(value)
+		if urlUnescape {
+			key = queryUnescape(key)
+			value = queryUnescape(value)
+		}
 		m[key] = append(m[key], value)
 	}
 	return m
 }
 
-// QueryUnescape is a non-strict version of net/url.QueryUnescape.
-func QueryUnescape(input string) string {
```

Comment:
Old comment about making it private: https://github.com/corazawaf/coraza/pull/326#discussion_r950272833

Reply:
I love when code becomes private.